tools/perf-regression-finder/src/SemanticAzureAi.cs

```diff
@@ -0,0 +1,163 @@
+                    another 2x to 3x increase in memory allocation. Additionally, repeated reallocations and inefficient buffer management magnify the memory overhead. These inefficiencies occur while serializing and buffering data in several intermediate steps, leading to high 
+                    memory consumption for large payloads.
+
+                    Call Stack:
+                    - User Code:
+                      at MyApp.OpenAiDemo.SendToGptAsync(Byte[] input)          (Program.cs:line 42)
+                        Invokes client.GetChatCompletionsStreamingAsync();
+                    - Azure.AI.OpenAI Layer:
+                      at Azure.AI.OpenAI.ChatCompletionsClient.GetChatCompletionsStreamingAsync(ChatCompletionsOptions options, CancellationToken cancellationToken) (OpenAIClient.g.cs:line 185)
+                        Starting API serialization
+                    - Azure.Core/Pipeline:
+                      at Azure.Core.Pipeline.HttpPipeline.SendAsync(RequestContent content, CancellationToken cancellationToken) (HttpPipeline.cs:line 128)
+                        Repetitively allocates buffers for encoding and escaping payloads
+                    - JSON/Serialization:
+                      at Azure.Core.Shared.Utf8JsonWriterExtensions.WriteBase64StringValue(Utf8JsonWriter writer, Byte[] value, String format) (Utf8JsonWriterExtensions.cs:line 50)
+                        Serializes Base64 payload, causing a significant increase in memory allocation
+                      at System.Text.Json.JsonDocument.Parse(ReadOnlyMemory<Byte> utf8Json, JsonDocumentOptions options) (JsonDocument.cs:line XYZ)
+                        Creates additional internal payload buffers
+
+                    Resolution:
+                    1. Avoid sending large binary payloads as Base64 by configuring the API client to accept binary content directly, bypassing the need for JSON wrapping and escaping.
+                    2. Optimize SDK serialization—modify or patch the SDK to prevent redundant escaping of Base64 strings, reducing memory overhead.
+                    3. Pre-allocate buffers close to the final content size within `ArrayBufferWriter` to minimize reallocations during serialization.
+                    4. If direct fixes are not possible, report the issues to the maintainers of `Microsoft.SemanticKernel` and `Azure.Core` to address inefficiencies with their serialization approach.
+
+
+                    """
+                );
+                // Update the appsettings.json with the new agent ID
+                UpdateAppSettings("./bin/Debug/net8.0/appsettings.json", "AzureAIFoundryAgentId", definition.Id);
+            }
+            // 2. Create plugins for the agent (Replace with your actual local paths)
+            var azureCore = new RepoAccessPlugin(@"C:\Users\t-akpoyibos\source\repos\azure-sdk-for-net\sdk\core\Azure.Core\src");
```

Comment:
Could the agent prompt for this at startup rather than it needing to be hardcoded?